### ISSUE TITLE
[Bug]: With gp3 storage_type for aws_db_instance cannot increase storage with gp3, or change to gp3, with storage < 400GB

### ISSUE THREAD

**Issue by craigmiskell-gitlab** [hashicorp/terraform-provider-aws#28160]
### Terraform Core Version

1.2.9

### AWS Provider Version

4.45.0

### Affected Resource(s)

aws_db_instance

### Expected Behavior

Should be able to make change to an RDS with gp3 storage (recently added) with < 400GB storage.

### Actual Behavior

If `iops` is not provided, the following error is thrown:

> api error InvalidParameterCombination: The storage type gp3 requires `iops` to be specified.

If the default 3000 IOPS for gp3 is provided in the `iops` attribute, but the storage size is smaller than 400GB, the following error is thrown:

> api error InvalidParameterCombination: You can't specify IOPS or storage throughput for engine postgres and a storage size less than 400.

Most interestingly, it's possible to create an RDS instance with gp3 and minimal storage (20GB) but increasing that (e.g. to 30GB) throws the above errors, and if the RDS already exists but with another storage type like gp2, it's not possible to upgrade to a gp3 in place.

### Terraform Configuration Files
```
terraform {
  # Note: this should match the version in `.tool-versions`
  required_version = "~> 1.2"

  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.9"
    }
    random = {
      source  = "hashicorp/random"
      version = "~> 3.3"
    }
  }
}

provider "aws" {
}


resource "aws_vpc" "vpc" {
  cidr_block           = "10.0.0.0/16" 
  enable_dns_support   = true
  enable_dns_hostnames = true
}

resource "aws_subnet" "public_subnet1" {
  vpc_id            = aws_vpc.vpc.id
  cidr_block        = "10.0.1.0/24"
  availability_zone = "ap-southeast-2a" 
}
resource "aws_subnet" "public_subnet2" {
  vpc_id            = aws_vpc.vpc.id
  cidr_block        = "10.0.2.0/24"
  availability_zone = "ap-southeast-2b" 
}

resource "aws_db_subnet_group" "rds" {
  name       = "rds"
  subnet_ids = [aws_subnet.public_subnet1.id, aws_subnet.public_subnet2.id]
}

resource "aws_db_instance" "rds" {
  identifier     = "mydb"
  engine         = "postgres"
  engine_version = "14.5" 
  instance_class = "db.m6g.large"
  multi_az       = true
  storage_type   = "gp3"

  db_name  = "mydb"
  username = "myuser"
  password = "abcdefgh" 

  db_subnet_group_name = aws_db_subnet_group.rds.name
  apply_immediately    = true

  allocated_storage     = 20
  max_allocated_storage = 1000
  storage_encrypted     = true
  skip_final_snapshot   = true
}
```

### Steps to Reproduce

Create the resources as described , then change allocated_storage to a larger number and apply, observe the failures as noted in the Actual Behavior section.

### References

Relates https://github.com/hashicorp/terraform-provider-aws/issues/27702.
Relates https://github.com/hashicorp/terraform-provider-aws/pull/27670.

https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Storage.html

### Would you like to implement a fix?

No

**Comment by gmiizi** [comment]
Hi,
We are blocked with this behaviour. Any chance to see it fixed soon ? ;)

Thx

**Comment by ewbankkit** [comment]
I can reproduce for MySQL with a slight modification of the existing acceptance test case:

```console
% make testacc TESTARGS='-run=TestAccRDSInstance_gp3MySQL' PKG=rds ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/rds/... -v -count 1 -parallel 3  -run=TestAccRDSInstance_gp3MySQL -timeout 180m
=== RUN   TestAccRDSInstance_gp3MySQL
=== PAUSE TestAccRDSInstance_gp3MySQL
=== CONT  TestAccRDSInstance_gp3MySQL
    instance_test.go:4761: Step 3/3 error: Error running apply: exit status 1
        
        Error: updating RDS DB Instance (tf-acc-test-5977403606277891875): operation error RDS: ModifyDBInstance, https response error StatusCode: 400, RequestID: 6086d274-e75b-445c-afc5-8f6b14bbba02, api error InvalidParameterCombination: You can't specify IOPS or storage throughput for engine mysql and a storage size less than 400.
        
          with aws_db_instance.test,
          on terraform_plugin_test.tf line 15, in resource "aws_db_instance" "test":
          15: resource "aws_db_instance" "test" {
        
--- FAIL: TestAccRDSInstance_gp3MySQL (566.04s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/rds	571.102s
FAIL
make: *** [testacc] Error 1
```

**Comment by etiennechabert** [comment]
@ewbankkit maybe we were a bit quick to close this one... 

I am now blocked by the same issue with `mariadb` while trying with the AWS Provider version `4.46.0`:

```
  ~ resource "aws_db_instance" "this" {
        id                                    = "a_db"
      ~ iops                                  = 0 -> 3000
        name                                  = "a_db"
      ~ storage_type                          = "gp2" -> "gp3"
        tags                                  = {
```

Causing this error:
```
You can't specify IOPS or storage throughput for engine mariadb and a storage size less than 400
```

As said in other issues, I am not blocked by this error message anymore for my DBs running on `postgres`

**Comment by ewbankkit** [comment]
@etiennechabert Please open a new issue (and like to this one).
Thanks.
